lib/tasks/grda_warehouse.rake

```diff
@@ -325,13 +325,23 @@ namespace :grda_warehouse do
     TaskQueue.queue_unprocessed!
     GrdaWarehouse::ProjectGroup.maintain_project_lists!
 
+    # Run HMIS Auto-Exit daily in the early morning. This is running here instead of the daily tasks because of the daily task is bloated.
     if DateTime.current.hour == 5 && HmisEnforcement.hmis_enabled? && GrdaWarehouse::DataSource.hmis.exists?
-      # Run HMIS Auto-Exit daily in the early morning. This is running here instead of the daily tasks because of the daily task is bloated.
-      Hmis::AutoExitJob.perform_now
+      begin
+        Hmis::AutoExitJob.perform_now
+      rescue StandardError => e
+        Sentry.capture_exception(e)
+        Rails.logger.error(e.message)
+      end
```

Comment:
This seems reasonable to capture errors but not prevent subsequent tasks from running.  An alternative, if timing isn't critical, would be simply to use `perform_later` so it runs in its own DJ worker.

Reply:
Some of these jobs that are running synchronously in this task might be pretty slow. For example:
* `driver:hmis:process_activity_logs`  - runs hmis activity log processing synchronously
* `PurgeSoftDeletedRecordsJob`  - when we turn this on in prod the first time it will be huge
* `AutoExitJob` (usually fast but could be slow if new auto exit rule is enabled for a project with lots of stale enrollments)


I think we can decouple this question from the PR, but it seems to me that those should be queued as async jobs instead of being run in-line in this task. Let me know if you agree. I would probably make that another PR since its more impactful.

Reply:
If they aren't time-sensitive, I say we adjust to `perform_later` (another PR is fine.)

Once we get to EKS, my understanding is that cron jobs are less expensive (they can be kicked off by a single watcher instead of separate ECS scheduled tasks), so we should revisit what is run on a schedule vs async.  

That said, maybe we want to wait to adjust these if we think everything's working fine now.